src/udiskslinuxdriveata.c

```diff
@@ -461,9 +480,12 @@ static gboolean get_pm_state (UDisksLinuxDevice *device, GError **error, guchar
       g_prefix_error (error, "Error sending ATA command CHECK POWER MODE: ");
       goto out;
     }
-  /* count field is used for the state, see ATA8: table 102 */
-  *count = output.count;
-  rc = TRUE;
+  /* state field is used for the state, see ATA8: table 102 */
+  *state = output.count;
+
+  if (state == 0xFF || state == 0x80)
```

Comment:
Based on the documentation provided above the test should probably look like `state > 0` for the _active_ state be assumed. Please check with the ATA specification but I guess the three values noted above are not the only ones possible. This would most likely vary across vendor implementations.